src/components/measurements/measurementsD3.js

```diff
@@ -365,3 +368,36 @@ export const addHoverPanelToMeasurementsAndMeans = (ref, handleHover, treeStrain
     })
     .on("mouseout.hoverPanel", () => handleHover(null));
 };
+
+export const addColorByAttrToGroupingLabel = (ref, treeStrainColors) => {
+  const svg = select(ref);
+  // Remove all previous color-by labels for the y-axis
+  svg.selectAll(`.${classes.yAxisColorByLabel}`).remove();
+  // Loop through the y-axis labels to check if they have a corresponding color-by
+  svg.selectAll(`.${classes.yAxis}`).select(".tick")
+    .each((_, i, elements) => {
+      const groupingLabel = select(elements[i]);
+      const groupingValue = groupingLabel.text();
+      const groupingValueColorBy = treeStrainColors[groupingValue];
```

Comment:
These colours are only set if a reference strain is visible - for instance change the date range slider and notice the reference strain colours start to disappear. I haven't walked up the code here to find out why this is, but ideally the colours should be here regardless of the visibility of the underlying reference strain. If this proves hard to implement we could add it as a to-do to #1463

Reply:
Ah thanks for pointing that out! This should be easy to tweak. I filtered colors to only visible strains since we only display visible strains for the measurements, but I can remove the filter.